contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/look_up_xtest.py

```diff
@@ -1,31 +1,33 @@
 #!/usr/bin/env python3
 
 from checkbox_support.snap_utils.snapd import Snapd
-from checkbox_support.snap_utils.system import get_gadget_snap
+import os
 
 
-def look_up_xtest():
-    if Snapd().list("x-test"):
-        return "x-test.xtest"
-    elif look_up_gadget() is not False:
-        return look_up_gadget()
-    else:
-        raise SystemExit(1)
-
-
-def look_up_gadget():
-    gadget = get_gadget_snap()
-    snap = Snapd().list(gadget)
-    if "apps" in snap.keys():
-        for app in snap["apps"]:
-            if app["name"] == "xtest":
-                return ".".join([app["snap"], app["name"]])
-    return False
+class ExtendSanpd(Snapd):
+    _apps = "/v2/apps"
 
+    def __init__(self, task_timeout=30, poll_interval=1, verbose=False):
+        super().__init__(task_timeout, poll_interval, verbose)
 
-def main():
-    print(look_up_xtest())
+    def list_apps(self, snaps):
+        return self._get(self._apps, params={"names": snaps})
 
 
-if __name__ == "__main__":
-    main()
+def look_up_xtest():
+    """Lookup xtest and tee-supplicant apps."""
+    a = ExtendSanpd()
+    snap = os.environ.get("XTEST")
+    apps = a.list_apps(snap)
+    results = {"xtest": None, "tee": None}
+    if apps["result"]:
+        for app in apps["result"]:
+            if app["name"] == "xtest":
+                results["xtest"] = "{}.xtest".format(app["snap"])
+            if app["name"] == "tee-supplicant":
+                results["tee"] = "{}.tee-supplicant".format(app["snap"])
```

Comment:
I think we could simply filter out the apps likes this.
```
xtest_apps = {(app["snap"], app["name"]) for app in apps["result"] if app["name"] in ["tee-supplicant", "xtest"]}
If len(xtest_apps) > 2:
  # raise issue here due to more than one xtest snap inf the system
if xtest_apps[0][0] != xtest_apps[1][0]
  # raise issue here due to the snap name is not match  
If  xtest_apps[0][1] == xtest_apps[1][1]
  # raise issue here due to the app name is the same
```

Reply:
Pushed the fix